microsoft-edge/webview2/how-to/winrt-js-conversion.md

```diff
@@ -0,0 +1,139 @@
+---
+title: How WinRT types and members are represented in JavaScript
+description: How to use WinRT types and members from within web-side JavaScript code in a WebView2 app, when calling native-side WinRT code from web-side code.
+author: MSEdgeTeam
+ms.author: msedgedevrel
+ms.topic: conceptual
+ms.prod: microsoft-edge
+ms.technology: webview
+ms.date: 12/02/2022
+---
+# How WinRT types and members are represented in JavaScript
+
+Here's how to use WinRT types and members from within web-side JavaScript code in a WebView2 app, when calling native-side WinRT code from web-side code.
+
+
+<!-- ====================================================================== -->
+## Language equivalents
+
+The WebView2 WinRT JS Projection tool (**wv2winrt**) converts from WinRT to JavaScript language constructs as follows.
+
+| WinRT language construct | JavaScript representation | Notes |
+|---|---|---|
+| `UInt8`, `Int16`, `UInt16`, `Int32`, `UInt32`, `Int64`, `UInt64`, `Single`, `Double` | `Number` | |
+| `Char`, `String` | `String` | A JavaScript `String` instance is converted to a WinRT `String` instance. |
+| `Boolean` | `Boolean` | |
+| `Windows.Foundation.DateTime` struct | `Date` | |
+| `Windows.Foundation.TimeSpan` struct | `Number` | |
+| `Guid` | `String` | A JavaScript `String` instance that contains a string representation of a UUID (with or without delimiting `{` and `}` braces) is converted to the corresponding UUID.  A UUID is converted to its string representation, with delimiting `{` and `}` brace characters at the start and end.  For information about UUID, see [RFC 4122](https://www.rfc-editor.org/rfc/rfc4122). |
+| `IVector<T>`, `IVectorView<T>`, `IObservableVector<T>` | `Array` and JavaScript object | If a `RuntimeClass` instance implements `vector` interfaces, then it is represented in JavaScript as the usual object described below, but also will act like a JavaScript array.  Reads and writes are performed live on the underlying WinRT vector object.
+| `IMap<K,V>`, `IMapView<K,V>`, `IObservableMap<K,V>` | JavaScript object | If a `RuntimeClass` instance implements `map` interfaces, then it is represented in JavaScript as the usual object described below, but also has properties with name and values from the underlying WinRT map object.  Reads and writes are performed live on the underlying WinRT map object. |
+| `Enum` | JavaScript object | An enum type is represented as a JavaScript object.  Each enum value is a `Number` property on the JavaScript object. |
+| `Struct` | JavaScript object | A `Struct` type is converted to a JavaScript object that has property names which correspond to the `Struct` type member names.  This is a two-way conversion. |
+| `Namespace` | JavaScript object | A namespace is represented as a JavaScript object that has a property for each child namespace, enum type, and `RuntimeClass`. |
```

Comment:
It feels like 'child namespace, enum type, and runtimeclass" should all be plural. A namespace may contain none or some child namespaces, child enum types, and child runtimeclass types.

```suggestion
| `Namespace` | JavaScript object | A namespace is represented as a JavaScript object that has a property for each child namespace, enum type, and `RuntimeClass`. |
```

Reply:
```
| `Namespace` | JavaScript object | A namespace is represented as a JavaScript object that has a property for any child namespaces, enum types, or `RuntimeClass`.  The namespace may have 0, 1, or many child namespaces, enums, or runtimeclasses, and each individual child namespace, enum and runtimeclass gets its own property. |
```